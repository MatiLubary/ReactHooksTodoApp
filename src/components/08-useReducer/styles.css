* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

body {
    padding: 50px;
}

p {
    margin: 0;
    cursor: pointer;
}

.list-group-item {
    display: flex;
    align-items: center;
    justify-content: space-between;
}

.mRa {
    margin-right: auto
}

.complete {
    text-decoration: line-through;
}

.w100 {
    width: 100%;
}