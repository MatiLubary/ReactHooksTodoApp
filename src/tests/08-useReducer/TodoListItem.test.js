import { shallow } from 'enzyme'
import React from 'react'
import { TodoListItem } from '../../components/08-useReducer/TodoListItem'
import { demoTodos } from "../fixtures/demoTodos";


//demoTodos

describe('tests with <TodoListItem />', () => {

    const handleDelete = jest.fn();
    const handleToggle = jest.fn();



/*     let wrapper;

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setTodos={setTodos} />)


    })
 */

    const wrapper = shallow(
        <TodoListItem
            todo = {demoTodos[0]}
            index = {0}
            handleDelete={handleDelete}
            handleToggle={handleToggle}
        />
    )



    test('should display properly', () => {
        expect(wrapper).toMatchSnapshot()
        
    })

    test('should call the delete function', () => {

        wrapper.find('button').simulate('click');
        expect( handleDelete ).toHaveBeenLastCalledWith(demoTodos[0].id)
    })

    test('should call the toggle function', () => {
        wrapper.find('p').simulate('click');
        expect( handleToggle ).toHaveBeenLastCalledWith(demoTodos[0].id)
    })

    test('should show the text correctly', () => {
       
        const p = wrapper.find('p')
        expect( p.text().trim() ).toBe(`1. ${demoTodos[0].desc}`)
        
    })


    test('should have the class complete if the todo is done TODO.done = true', () => {
       
        const todo = demoTodos[0];
        todo.done = true;

        const wrapper = shallow(
            <TodoListItem
                todo = {todo}
            />
        )

        expect(wrapper.find('p').hasClass('complete')).toBe(true)



        
    })
    




    
    
    
})
