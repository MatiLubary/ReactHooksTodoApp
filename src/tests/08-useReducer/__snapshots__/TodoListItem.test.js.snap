// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests with <TodoListItem /> should display properly 1`] = `
<li
  className="list-group-item border-bottom"
  key="1"
>
  <p
    className=""
    onClick={[Function]}
  >
    1
    . 
    learn react
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
